.github/workflows/dockerfile-push.yml

```diff
@@ -24,8 +22,25 @@ jobs:
         username: ${{ secrets.DOCKERHUB_USERNAME }}
         password: ${{ secrets.DOCKERHUB_TOKEN }}
 
+    - name: Set up Docker Buildx
+      uses: docker/setup-buildx-action@v1
 
+    # Cache Docker layers
+    - name: Cache Docker layers
+      uses: actions/cache@v3
+      with:
+        path: /tmp/.buildx-cache
+        key: ${{ runner.os }}-buildx-${{ github.sha }}
```

Comment:
This creates individual docker caches over time. Better to use cache/restore and cache/save manually rather than relying on this.

Reply:
GitHub Actions will periodically clean up older cache entries based on usage and the retention policy, so it's not an infinite accumulation.

Reply:
Yep just create one key for cache for main. I think that will do the trick.

Reply:
Yup, completely agree with @Dsantra92 on that. Although in @NripeshN defense: This cache will only work during weekends, where we don't push commits to github. Basically building the image immediately.